metadata-ingestion/src/datahub/ingestion/source/sql/postgres.py

```diff
@@ -152,6 +157,23 @@ def create(cls, config_dict, ctx):
         config = PostgresConfig.parse_obj(config_dict)
         return cls(config, ctx)
 
+    @staticmethod
+    def test_connection(config_dict: dict) -> TestConnectionReport:
```

Comment:
This logic should go in SQLAlchemySource, not here

Ideally, it should work across all SQL-based sources via inheritance